project-generation/helpers.go

```diff
@@ -8,46 +8,91 @@ import (
 	"io"
 	"os"
 	"os/exec"
+	"strconv"
 	"strings"
 	"time"
 )
 
-// validateArguments will ensure that all user input via flags have been provided and if not prompt for them to be and
-// keep prompting until all input meets validation criteria
-func validateArguments(ctx context.Context, repositoryCreated bool, unvalidatedName, unvalidatedType, unvalidatedProjectLocation, unvalidatedGoVersion string) (string, string, string, string, error) {
-	var validatedAppName, validatedProjectType, validatedProjectLocation, validatedGoVersion string
-	var err error
-	validatedAppName, err = ValidateAppName(ctx, unvalidatedName)
-	validatedProjectType, err = ValidateProjectType(ctx, unvalidatedType)
+type ListOfArguments map[string]*Argument
+type Argument struct {
+	Validator func(context.Context, string) (string, error)
+	Context   context.Context
+	InputVal  string
+	OutputVal string
+}
+
+func configureAndValidateArguments(ctx context.Context, appName, projectType, projectLocation, goVersion string) (an, pt, pl, gv string, err error) {
+	listOfArguments := make(ListOfArguments)
+	listOfArguments["appName"] = &Argument{
+		InputVal:  appName,
+		Context:   ctx,
+		Validator: ValidateAppName,
+	}
+	listOfArguments["projectType"] = &Argument{
+		InputVal:  projectType,
+		Context:   ctx,
+		Validator: ValidateProjectType,
+	}
+	listOfArguments["projectLocation"] = &Argument{
+		InputVal:  projectLocation,
+		Context:   ctx,
+		Validator: ValidateProjectLocation,
+	}
+	listOfArguments, err = ValidateArguments(listOfArguments)
 	if err != nil {
+		log.Event(ctx, "validation error", log.Error(err))
 		return "", "", "", "", err
 	}
-	offerPurge := !repositoryCreated
-	validatedProjectLocation, err = ValidateProjectLocation(ctx, unvalidatedProjectLocation, validatedAppName, offerPurge)
+	an = listOfArguments["appName"].OutputVal
+	pt = listOfArguments["projectType"].OutputVal
+	pl = listOfArguments["projectLocation"].OutputVal
+	listOfArguments = make(ListOfArguments)
+	goVerUnset := goVersion == "unset" || goVersion == ""
+	if goVerUnset && projectType != "generic-project" {
+		listOfArguments["goVersion"] = &Argument{
+			InputVal:  goVersion,
+			Context:   ctx,
+			Validator: ValidateGoVersion,
+		}
+		gv = listOfArguments["goVersion"].OutputVal
+	}
+	listOfArguments, err = ValidateArguments(listOfArguments)
 	if err != nil {
+		log.Event(ctx, "validation error", log.Error(err))
 		return "", "", "", "", err
 	}
 
-	if unvalidatedGoVersion == "unset" && validatedProjectType != "generic-project" {
-		validatedGoVersion, err = promptForInput(ctx, "Please specify the version of GO to use")
+	return an, pt, pl, gv, nil
+}
+
+func ValidateArguments(arguments map[string]*Argument) (validArguments map[string]*Argument, err error) {
+	fmt.Println("function ValidateArguments hit")
+	validArguments = make(ListOfArguments)
+	for key, value := range arguments {
```

Comment:
I don't understand what this is achieving - we're populating a local map with the arguments as passed in, and then in the next loop we're looping through them all and updating them in the local map to return that as an updated list. Line 78 could be `value.OutputVal, err`, and then `validArguments` is entirely superfluous no?